macOS/DuckDuckGo/NetworkProtection/AppTargets/BothAppTargets/VPNUIActionHandler.swift

```diff
@@ -76,4 +90,45 @@ extension VPNUIActionHandler: VPNUIActionHandling {
     func showPrivacyPro() async {
         await vpnURLEventHandler.showPrivacyPro()
     }
+
+    @MainActor
+    func willStopVPN() async -> Bool {
+        guard vpnAppState.isUsingSystemExtension && !vpnAppState.dontAskAgainExclusionSuggestion,
```

Comment:
The alert will only be shown if the user is using a system extension (since that's the only case in which VPN exclusions work), and if the "don't ask again" checkbox hasn't been ticked in the past.